commit 793ae8c2b4cc651d9c6670bb03ebb0740987f4f6
Author: Ryan May <rmay@ucar.edu>
Date:   Wed Aug 25 16:29:45 2021 -0600

    Revert "Simply version check logic"
    
    This reverts commit e755a08f11dc7a3b31039a912334505bdd90ecd8.

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e1b620fee..dc95eeae9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -818,11 +818,19 @@ IF(USE_HDF5)
 
   # Check to see if HDF5 library is 1.10.6 or greater.
   # Used to control path name conversion
-  IF(${HDF5_VERSION} VERSION_GREATER "1.10.5")
-    SET(HDF5_UTF8_PATHS ON)
-  ELSE()
-    SET(HDF5_UTF8_PATHS OFF)
-  ENDIF()
+  # Check HDF5 version
+  SET(HDF5_UTF8_PATHS OFF)
+  CHECK_C_SOURCE_COMPILES("
+  #include <H5public.h>
+  int main() {
+#if (H5_VERS_MAJOR*10000 + H5_VERS_MINOR*100 + H5_VERS_RELEASE < 11006)
+      choke me;
+#endif
+  }" HDF5_VERSION_1106)
+
+  IF(HDF5_VERSION_1106)
+   SET(HDF5_UTF8_PATHS ON)
+  ENDIF(HDF5_VERSION_1106)
 
   MESSAGE("-- Checking for HDF5 version 1.10.6 or later: ${HDF5_UTF8_PATHS}")
 
diff --git a/nc_test4/CMakeLists.txt b/nc_test4/CMakeLists.txt
index c5c345005..d4f97bfd7 100644
--- a/nc_test4/CMakeLists.txt
+++ b/nc_test4/CMakeLists.txt
@@ -47,10 +47,6 @@ ENDIF(ENABLE_FILTER_TESTING)
 
 ENDIF(BUILD_UTILITIES)
 
-IF(${HDF5_VERSION} VERSION_GREATER "1.10.0")
-  SET(NC4_TESTS ${NC4_TESTS} tst_virtual_datasets)
-ENDIF(${HDF5_VERSION} VERSION_GREATER "1.10.0")
-
 ##
 # The shell script, run_empty_vlen_test.sh,
 # depends on the 'tst_empty_vlen_unlim' binary.
